### DISCUSSION TITLE
Status of support for third-party hdf plugins in NetCDF-Java

### DISCUSSION THREAD

**Discussion by mcucchi9** [Unidata/netcdf-java#922] [Questions and Answers] [answered]
Hi,

I'm trying to use the FCIDECOMP decompression filter with netCDF-Java based applications (such as ToolsUI). I know this can be easily done with netCDF-C based applications (thanks to the HDF5-C dynamically loaded filters support), but I'm not aware of any straightforward way to do it with netCDF-Java based ones.

For this reason, I was wondering if any further development of the netCDF-Java library is planned to enable support for dynamically
loaded filters and, if not, if you can point me to any workaround to allow such integration.

Thanks,
Marco

**Comment by haileyajohnson** [comment, marked as answer]
Hi Marco,

Dynamically loaded filters will be supported in the next release of netCDF-java, via a filter service provider. Once it's available, you'll be able to plug-in any filter, as long as you can wrap it in a Java class that extends the [Filter](https://github.com/Unidata/netcdf-java/blob/maint-5.x/cdm/core/src/main/java/ucar/nc2/filter/Filter.java) class. More documentation on how to provide third party filters will be available with next release as well.

Thanks!
